pennylane/operation.py

```diff
@@ -604,7 +604,12 @@ def matrix(self, wire_order=None):
         """
         canonical_matrix = self.compute_matrix(*self.parameters, **self.hyperparameters)
 
-        if wire_order is None or self.wires == Wires(wire_order):
+        # Linting check disabled as static analysis can misidentify qml.ops as the set instance qml.ops.qubit.ops
+        if (
+            wire_order is None
+            or self.wires == Wires(wire_order)
+            or self.name in qml.ops.qubit.attributes.symmetric_over_all_wires
```

Comment:
```suggestion
            or (self.name in qml.ops.qubit.attributes.symmetric_over_all_wires and set(self.wires) == set(wire_order))
```

For a case like `qml.IsingXX(1.234, wires=(0,1)).matrix(wire_order(2,1,0))`, we need to remember the existence of identity wires.